### PR TITLE
EVG-13803 dequeue item in git.merge

### PR DESCRIPTION
**Pull request by john-m-liu**
- make an extra API request at the end of git.push to dequeue the commit queue item (this is what git.merge_pr does already)
- delete the notification code that was doing the above on the app servers
Note this is technically not backwards compatible, but I believe it's fine to do this. If the timing is just right so that there is a CLI-initiated commit queue task that has finished merging, but the notification system has not yet gotten to the dequeue before the deploy, we'll potentially get some errors from the notification system not knowing how to handle the subscriber (because I delete the code here) and the unsticking part of the commit queue job will clean things up

### CODE REVIEW

File: agent/command/git_push.go [hunk 1]
```diff
@@ -47,11 +49,24 @@ func (c *gitPush) ParseParams(params map[string]interface{}) error {
 
 // Execute gets the source code required by the project
 func (c *gitPush) Execute(ctx context.Context, comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
-	if err := util.ExpandValues(c, conf.Expansions); err != nil {
+	var err error
+	defer func() {
+		pErr := recovery.HandlePanicWithError(recover(), nil, "unexpected error in git.push")
```
Comment (ablack12) [inline agent/command/git_push.go:54]: Do we expect something in here to be panicky?

File: model/notification/notification.go [hunk 2]
```diff
@@ -290,16 +282,14 @@ func (n *Notification) SetTaskMetadata(ID string, execution int) {
 }
 
 type NotificationStats struct {
-	GithubPullRequest  int `json:"github_pull_request" bson:"github_pull_request" yaml:"github_pull_request"`
-	JIRAIssue          int `json:"jira_issue" bson:"jira_issue" yaml:"jira_issue"`
-	JIRAComment        int `json:"jira_comment" bson:"jira_comment" yaml:"jira_comment"`
-	EvergreenWebhook   int `json:"evergreen_webhook" bson:"evergreen_webhook" yaml:"evergreen_webhook"`
-	Email              int `json:"email" bson:"email" yaml:"email"`
-	Slack              int `json:"slack" bson:"slack" yaml:"slack"`
-	GithubMerge        int `json:"github_merge" bson:"github_merge" yaml:"github_merge"`
```
Comment (ablack12) [inline model/notification/notification.go:299]: You removed the GithubMerge Notification stat but didn't remove it from the subscriber types; is this intentional?

**Review by ablack12 — CHANGES_REQUESTED** [review]
Just to check, if we were to revert this commit, commit queue items would still be set up correctly to go back to the notification system, right?

File: agent/command/git_push.go (hunk 1 above)
Comment (john-m-liu) [reply agent/command/git_push.go:54]: no, it's just a safety measure to try and make sure the code in the defer isn't affected by a potential panic

File: model/notification/notification.go (hunk 2 above)
Comment (john-m-liu) [reply model/notification/notification.go:299]: there's a todo to remove it at the end of the project. The comment actually wasn't worded such that I could find it, but I changed it

**Review by john-m-liu — COMMENTED** [review]
That's a good point, reverting while there are items in the queue would have the same small problem as the forward deploy. They won't dequeue themselves at the end, but the unsticking part of the job will clean them up

**Review by ablack12 — APPROVED** [review]

**Comment by john-m-liu** [comment]
evergreen merge

test

**Pull request merged**